ecommerce/admin.py

```diff
@@ -494,6 +497,39 @@ class DataConsentUserAdmin(TimestampedModelAdmin):
     model = DataConsentUser
 
 
+class DataConsentAgreementForm(forms.ModelForm):
+    """Form for DataConsentAgreementAdmin"""
+
+    class Meta:
+        model = DataConsentAgreement
+        fields = "__all__"
+
+    def clean(self):
+        is_global = self.cleaned_data.get("is_global", False)
+        courses = self.cleaned_data.get("courses", Course.objects.none())
+        company = self.cleaned_data.get("company", None)
+        # Check if a global agreement for a specific company already exists.
+        # (Only applicable if a new object is created or company is changed in an existing object)
+        if (
+            is_global
+            and DataConsentAgreement.objects.filter(company=company, is_global=True)
+            .exclude(pk__in=[self.instance.pk or -1])
```

Comment:
I added this because `self.instance.pk` in case of a new object is `None` and doing `.exclude(pk__in=[None])` would return 0 results. But the approach you mentioned without using the list filter works fine. Thanks for pointing this out.

I've updated the PR.